Cal/fatlib migration

CalNightingale opened the pull request:
Adds SD card data support

mlp6 committed:
initial integration of SD card data writing (issue #210)

mlp6 committed:
add SD library (issue #212)

mlp6 committed:
fix relative paths of SD library (and sublibraries) (issue #212)

mlp6 committed:
update local include path (issue #212)

mlp6 committed:
fix define preprocessor syntax

mlp6 committed:
rm spurious compiled firmware

mlp6 committed:
move SD card pin definition to pin_mapping.h

mlp6 committed:
update pin mapping

mlp6 committed:
Merge branch 'master' into mlp6/sdcard

CalNightingale committed:
Attempt to add time to data_str

CalNightingale committed:
Revised conditional time addition to data_str

CalNightingale committed:
Revised addition of timestamp V2

Added if/else statement to determine whether to include the timestamp based on the presence of an SD card

CalNightingale committed:
[WIP] Refactoring of publish function into a class

CalNightingale committed:
[WIP] Publish() refactoring

CalNightingale committed:
[WIP] Move vars that are published to a new file

CalNightingale committed:
[WIP] Publish() refactoring (addresses issue #214)

mlp6 committed:
ignore clion-log.txt

mlp6 committed:
WIP: pass input vars to cell publish method

CalNightingale committed:
Merge branch 'cal/sdcard' into mlp6/cal/sdcard

CalNightingale committed:
Merge pull request #217 from mlp6/mlp6/cal/sdcard

pass input vars to cell publish method

CalNightingale committed:
Refactored publish() (fixes issues #217, #216, #215, #214)

CalNightingale committed:
Added separate bools to keep track of publish() status

CalNightingale committed:
Import of required libraries, misc code changes

CalNightingale committed:
Defined all pins for SD card

CalNightingale committed:
Changed File to FatFile

CalNightingale committed:
Added logs to _pushed() functions

CalNightingale committed:
Added more logs

CalNightingale committed:
Added logs in loop() method

CalNightingale committed:
Added logs to publish and setup() methods

CalNightingale committed:
changed SPI mode

CalNightingale committed:
Add logHandler initialization

CalNightingale committed:
Lowered log level for read_temp()

CalNightingale committed:
removed PublishSD.h, placed in main file

CalNightingale committed:
Reintegrated all loop() functions except cellPublish()

CalNightingale committed:
Reintegrated all loop() functions except cellPublish()

CalNightingale committed:
Merge branch 'cal/fatlibMigration' of https://github.com/mlp6/ADPL into cal/fatlibMigration

CalNightingale committed:
Reintegrated temp_read() and _pushed() functions

CalNightingale committed:
Reimplemented cellPublish() and related info

CalNightingale committed:
Changed thermistor pins, NOW FUNCTIONAL

CalNightingale committed:
Reverted thermistor pins, switched to SPI3 config

CalNightingale committed:
Changed SD pins to B0-B3

CalNightingale committed:
Reimplemented PublishDataSD.h

CalNightingale committed:
Removed testing code

CalNightingale committed:
Removed more test code, reimplemented variable setup

CalNightingale committed:
Added variable determining existence of card, removed unnecessary files

CalNightingale committed:
Fixed variable determining SD card presence

CalNightingale committed:
Added updater for publishedSD bool

CalNightingale committed:
Replaced chipSelect with definition in pin_mapping.h

Also added definitions for all SD pins and removed some unnecessary comments

CalNightingale committed:
Replaced all SD definitions with pin_mapping.h

CalNightingale committed:
Implemented exhaust thermistor

Replaced 15 minute ignition delay with data from the exhaust thermistor, added the exhaust thermistor to the read_temp() method

CalNightingale committed:
Lowered temp threshold, added delay between firings

mlp6 committed:
Merge branch 'master' into cal/fatlibMigration

mlp6 commented on software/ADPL_electron/ADPL_electron.ino:
```diff
@@ -10,13 +10,29 @@
+File sdFile;
+PublishDataSD sdPublisher;
+#else
+#include "pin_mapping.h"
```
@CalNightingale it appears that `pin_mapping.h` is included regardless of the value of `sdIsInstalled`; can we move that outside the `if/else` structure, or is that required to get things to compile?  I would guess it should be able to be moved out... was this the error that you encountered before?

CalNightingale replied:
Yes. Because (no idea why) an error is thrown if `pin_mapping.h` is included before including both publishing methods, this redundancy is necessary

mlp6 commented on software/ADPL_electron/ADPL_electron.ino:
```diff
@@ -59,7 +76,18 @@ unsigned long last_publish_time = 0;
 int write_address = 0;
 
+//TEST VAR
+bool sDSuccess = false;
```
Do we still need this variable?

CalNightingale replied:
Nope! I have removed it and made another commit.

CalNightingale committed:
Added logic based on temp increase

CalNightingale committed:
Removed test variable sDSuccess

CalNightingale committed:
Removed unnecessary files

CalNightingale committed:
[WIP] added pushToCell function

CalNightingale committed:
[WIP] Refactored file reading in pushToCell function

Now utilizeses fgets function of file class, instead of char by char reading

CalNightingale committed:
Merge branch 'master' into cal/fatlibMigration

CalNightingale committed:
Finalized pushToCell function

Added code to delete cached data if it is successfully pushed to the server

CalNightingale committed:
Replaced millis() with Time() functions

These sync with the server whenever there's a connection

CalNightingale committed:
Finalized SD pins

They're now in ascending order, making wiring much easier

CalNightingale committed:
Only publishes to SD if no cell connection

CalNightingale committed:
Added logError function

CalNightingale committed:
added error code file, implemented on all Log.error() calls

CalNightingale committed:
Merge branch 'cal/fatlibMigration' into cal/errorCodes

CalNightingale committed:
Merge pull request #245 from mlp6/cal/errorCodes

Cal/error codes

CalNightingale committed:
Implemented logError() method

Added all necessary error codes to error_codes.h

CalNightingale committed:
Adds log to SD functionality

CalNightingale committed:
Reformatted timestamps into UTC

Addresses issue #239

mlp6 committed:
rm log.() calls (firmware too large)

mlp6 committed:
add Graham's latest pin mappins

mlp6 committed:
integrate the latest batch flow logic changes from Graham

mlp6 committed:
rm unused variable

mlp6 committed:
fix some logic for the pinch valve (issue #250)

mlp6 committed:
try to more robustly deal with hot plug/remove (issue #252)

mlp6 committed:
fix pinch valve login (issue #250)

* pinchValve attributes up/down renamed to moveup/movedown to clarify
that they indicate intention to move, not state of valve
* isitup attribute added to indicate the actual state of the valve,
which is not just a binary up/down

mlp6 committed:
redefine RESET button action per Graham

mlp6 committed:
refactor global WAIT_TIME to pinchValve.wait_time attribute

mlp6 reviewed (commented):
Waiting for Graham's testing on the device before merging in.

mlp6 committed:
minor refactor

mlp6 committed:
fix EXH pin mapping; small refactors

mlp6 committed:
update CHANGELOG for v2.5.0RC1; update contributors

mlp6 committed:
fix epoch timestamp (#256)

fix epoch timestamp writing (closes #253)

mlp6 committed:
Mlp6/exhaust counter limit (#257)

* make delay after opening gas value part of valve.open()

* replace global IGNITOR_DELAY with ignitor.refireDelay attribute

* add ignitor refire lockout (closes #254)

mlp6 committed:
update CHANGELOG

mlp6 committed:
Merge branch 'master' into cal/fatlibMigration

mlp6 reviewed (approved):
Graham has tested all of these changes on the current hardware setup going to the Philippines.
